### PR TITLE
Migration to vue 3.

### PR DESCRIPTION
**Pull request by alan-wu**

### CODE REVIEW

File: vite.config.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import Components from 'unplugin-vue-components/vite'
+import { ElementPlusResolver } from 'unplugin-vue-components/resolvers'
+
+// https://vitejs.dev/config/
+export default defineConfig(({ command, mode }) => {
+  const config = {
+    css: {
+        preprocessorOptions: {
+          scss: {
+            additionalData: `@use '${pathSrc}/assets/styles' as *;`
```
Comment (Tehsurfer) [inline vite.config.js:14]: This doesn't run for me, I need to switch it to:

```
additionalData: `@use './assets/styles' as *;`
```

**Review by Tehsurfer — APPROVED** [review]
Thats a lot of changes! I read through them all and they all look good. 

I had a few small questions, and also it doesnt run for me unlesse I change the config (maybe windows things :( )

File: src/components/MapContent.vue [hunk 2]
```diff
@@ -13,35 +12,33 @@
...
-import Vue from "vue";
-import SplitFlow from './SplitFlow';
+import SplitFlow from './SplitFlow.vue';
 import EventBus from './EventBus';
-import store from '../store';
+import { mapStores } from 'pinia';
+import { useSettingsStore } from '../stores/settings';
 import { findSpeciesKey } from './scripts/utilities.js';
-import {MapSvgSpriteColor} from '@abi-software/svg-sprite';
+import { MapSvgSpriteColor} from '@abi-software/svg-sprite';
```
Comment (ddjnw1yu) [inline src/components/MapContent.vue:27]: Is there any affect if not import the SVG style here? From what I met earlier during the migration, it does affect the style.

File: src/components/ContentBar.vue [hunk 3]
```diff
@@ -56,47 +57,67 @@
...
 </template>
 
 
 <script>
 /* eslint-disable no-alert, no-console */
-import Vue from "vue";
 import EventBus from './EventBus';
-import store from "../store";
-import ContextCard from "./ContextCard";
+import { MapSvgIcon } from '@abi-software/svg-sprite';
```
Comment (ddjnw1yu) [inline src/components/ContentBar.vue:87]: Same here.

**Review by ddjnw1yu — APPROVED** [review]
It looks good to me. Thanks! Quite a lot of them are new to me, and I have yet to get a chance to run this locally. I will return to familiarize myself with the changes once I have more time.

File: .env [hunk 4]
```diff
@@ -1,6 +1,6 @@
-//VUE_APP_API_LOCATION=https://your-api-location/
-VUE_APP_FLATMAPAPI_LOCATION=https://mapcore-demo.org/current/flatmap/v3/
-VUE_APP_ALGOLIA_INDEX=k-core_dev_published_time_desc
-VUE_APP_ALGOLIA_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
-VUE_APP_ALGOLIA_ID=xxxxxxxxx
-VUE_APP_PENNSIEVE_API_LOCATION=https://api.pennsieve.io
+//VITE_API_LOCATION=https://your-api-location/
+//VITE_FLATMAPAPI_LOCATION=https://mapcore-demo.org/current/flatmap/v3/
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply .env:2]: I comment them out as they are prioritised over .env.local in my systemfor some reason.

File: package.json [hunk 5]
```diff
@@ -1,16 +1,15 @@
 {
   "name": "@abi-software/mapintegratedvuer",
-  "version": "0.6.7",
+  "version": "0.7.0-vue3.7",
   "license": "Apache-2.0",
   "scripts": {
-    "serve": "vue-cli-service serve --port 8081",
-    "build": "vue-cli-service build --mode prod",
+    "serve": "vite --host --force",
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply package.json:6]: No, --force will force it to recreate the dependencies cache, it is useful for testing new dependencies (copy and paste methode).

File: src/assets/styles.scss [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
-
 @import '_variables';
 
+@import url('https://fonts.googleapis.com/css?family=Asap:400,400i,500,600,700&display=swap');
+
 /* icon font path, required */
 $--color-primary: $app-primary-color !default;
-$--font-path: '~element-ui/lib/theme-chalk/fonts';
+:root {
+  --el-color-primary: #8300BF;
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/assets/styles.scss:8]: I should remove that anyway, we set it from the mapcontent class.

File: src/components.d.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable */
+/* prettier-ignore */
+// @ts-nocheck
+// Generated by unplugin-vue-components
+// Read more: https://github.com/vuejs/core/pull/3399
+export {}
+
+declare module 'vue' {
```
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/components.d.ts:8]: I need to look into this.

File: src/components/MapContent.vue (hunk 2 above)
Comment (alan-wu) [reply src/components/MapContent.vue:27]: It does affect it if I remember correctly.

File: src/components/ContentBar.vue (hunk 3 above)
Comment (alan-wu) [reply src/components/ContentBar.vue:87]: In this case, it is use at line 72.

File: src/components/MapContent.vue (hunk 2 above)
Comment (ddjnw1yu) [reply src/components/MapContent.vue:27]: I run the vuer locally. Style looks not changed with/without the svg style imported. All good, thanks!

**Pull request merged**